src/modules/flow/update/update.c

```diff
@@ -30,20 +30,15 @@
 
 struct update_data {
     struct sol_update_handle *handle;
-};
-
-struct update_node_type {
```

Comment:
I do not agree with this one. You can still leave the progress_port at the update_node_type and do the following in the json spec. Check below.